src/main/scala/scala/tools/partest/PartestDefaults.scala

```diff
@@ -25,7 +25,9 @@ object PartestDefaults {
     import scala.reflect.io.Path, Path._
     Some(Path(jdkHome) / "bin") filter (_.isDirectory) flatMap { p =>
       val candidates = (p walkFilter { e => (e.name == name || e.name.startsWith(s"$name.")) && e.jfile.canExecute }).toList
-      (candidates find (_.name == name) orElse candidates.headOption) map (_.path)
+      val plain = candidates find (_.name == name)
+      val exe = candidates find (_.name == s"${name}.exe")
+      (plain orElse exe orElse candidates.headOption).map(_.path)
```

Comment:
In practice, if there is neither a `name` or `name.exe`, the guess is most likely just wrong. The theoretical platform/jdk/jre that uses the same name but a different extension probably doesn't exist, and abstracting over that hypothetical probably isn't worth the brain cycles.

replacing the whole thing with something like

`(p walkFilter { e => (e.name == name || e.name == e.name.looksLike(s"${name}.exe") && e.jfile.canExecute }).map(_.path).headOption`

or even just

`(Path(jdkHome)  / "bin" walkFilter { e => (e.name == name || e.name == s"${name}.exe") && e.jfile.canExecute }).map(_.path).headOption
`
should work fine enough. It's not going to find `java.woozle` if that is the correct file (which it isn't). If it's looking for a file named `bin` (which it isn't) *and* `jdkhome/bin` is a file rather than a directory (which it also isn't) it's going to return the path of `jdkhome/bin` rather than `None` mistakenly. I'm not sure that would be a great loss.